### DISCUSSION TITLE
[FE 레벨4] 유연하게 재사용 가능한 레이아웃 컴포넌트

### DISCUSSION THREAD

**Discussion by wmakerjun** [woowacourse/retrospective#43] [General]
### 테스트 및 리뷰할 때 아래 가이드라인을 참고해주세요.
- 여러 환경에서 잘 보이는가?
    - 반응형: 레이아웃이 다양한 화면 크기에 적응하는지 확인해 주세요.
    - 크로스 브라우징: 여러 브라우저(크롬, 사파리, 엣지 등)에서 동작하는지 확인해 주세요.
- 문서가 가독성 있게 쓰여있는가?
    - README나 사용법을 보고, 저자의 의도가 잘 이해되는지 독자의 입장에서 확인해 주세요.
    - 레이아웃 컴포넌트를 사용하는 입장에서 쉽게 사용할 수 있었는지 확인해 주세요.
- ‘유연’하게 재사용 하기 좋은가?
    - 고려한 사용자에게 발생할 수 있는 상황에 따른 props를 고려했는가?
        - 예를 들어 사용자 입장에서 웹페이지의 컬러에 맞게 Tab Component의 색상을 변경할 수 있는 속성이 있는지 확인해볼 수 있습니다.
    - 라벨 텍스트가 매우 긴 경우를 고려했는가?
    - 라벨에 아이콘이 있는 경우/없는 경우/같이 들어가는 경우를 고려했는가
    - 가로 외에 세로형으로도 보여야 하는 경우는 어떻게 보이는가?
    - 탭의 수가 많은 경우 어떻게 보여주는가?

**Comment by wmakerjun** [comment]
# 노아 - Tab Component
https://github.com/woowacourse/layout-component/pull/1

**Reply by wmakerjun** [reply]
# 긍정적인 리뷰

# 개선 제안

**Reply by woo-jk** [reply]
### 긍정적인 리뷰
- 화면에 짤렸을 때 스크롤 버튼 옵션 줄수 있는거 좋아요.
- 스크롤 버튼을 눌렀을 때 애니메이션도 정말 좋아요.
- 버티컬 탭 지원하는거 대박입니다.
- panel의 제목(label)은 프롭스로 받고, 요소들은 children으로 받을 수 있게 해주셨는데, 사용하기 쉽고 가독성도 좋아서 개발자 경험이 되게 좋을 것 같아요.
- 합성 컴포넌트 사용하기 때문에 코드 가독성이 좋아질 것 같아요.
- README도 이해하기 편했어요.

### 개선 제안
- 화면이 좁아질 때 스크롤 버튼을 굳이 옵션으로 넣어야하나? 라는 생각이 들어요. 옵션을 주지 않고 무조건 적용되면 좋을 것 같아요.
- 스크롤 버튼이 조금 작은 것 같아요. 클릭 혹은 터치하기가 힘든데 좀 더 키우면 좋을 것 같아요.
  - 그리고 버튼에 세로 중앙 정렬이 안되어있는 것 같아요!
- panel이 disabled 일 때 마우스 호버 시 마우스 효과를 주면 어떨까요?
- 마우스 드래그로도 스크롤을 할 수 있다면 사용성이 엄청 좋아질 것 같아요.

**Reply by gyeongza** [reply]
## 긍정적인 리뷰
- npm 문서가 잘 정리되어있어서 이해하기 쉬웠습니다.
- default 값을 주어 기본 속성으로 사용하기 편리했습니다.
- 사파리, 크롬, 엣지 브라우저에서 잘 동작합니다!

## 개선 제안


![Sep-26-2023 11-34-59](https://github.com/woowacourse/retrospective/assets/62369936/09fadd00-fa56-4cec-b3f1-dcbc399045a5)

- 첫 번째 라벨이 disabled라면 활성화가 되어있다가 다음 탭으로 넘어가면 그때 비활성화가 됩니다. 혹시 의도된건지 궁금합니다!

---

<img width="500" alt="스크린샷 2023-09-26 오전 11 35 50" src="https://github.com/woowacourse/retrospective/assets/62369936/661e5059-f9ec-4392-9432-9cada5e78807">

- npm README에 Tabs 컴포넌트는 반응형이 지원되지 않는다고 나와있는데 현재는 지원되는 건가요? 그러면 업데이트를 해야할 것 같습니당

---

![Sep-26-2023 11-46-25](https://github.com/woowacourse/retrospective/assets/62369936/f5938460-eec4-4786-8472-f3611daae9d5)

![Sep-26-2023 11-42-19](https://github.com/woowacourse/retrospective/assets/62369936/3e726096-88b9-455f-8d64-0b98a75b586c)

- Navigation의 가로폭이 화면 폭보다 길어졌을 때 스크롤이 나오는데 label의 글자 수가 적을 때는 어느정도 맞게 나오지만 길어지면 어느 기준으로 스크롤이 나오는걸까요?

**Reply by turtle601** [reply]
## 푸만능

### 긍정적인 리뷰
1. 모바일 접근 측면을 고려해서 Tab Scroll 기능까지 구현한 것이 너무 인상적이었어요!!
2. MUI 라이브러리에서 제공하는 기능들을 거의 모두 제공해준 것이 매우 인상적이었습니다!!
3. Tab의 label이 동일할 경우 컴파일 에러를 발생시키는 것이 인상적이었습니다!!

### 체크한 리뷰 항목
- [x] 반응형: 레이아웃이 다양한 화면 크기에 적응하는지 확인해 주세요.
- [x] 크로스 브라우징: 여러 브라우저(크롬, 사파리, 엣지 등)에서 동작하는지 확인해 주세요.

- [x] README나 사용법을 보고, 저자의 의도가 잘 이해되는지 독자의 입장에서 확인해 주세요.
- [x] 레이아웃 컴포넌트를 사용하는 입장에서 쉽게 사용할 수 있었는지 확인해 주세요. 고려한 사용자에게 발생할 수 있는 상황에 따른 props를 고려했는가?

- [x] 유연하게 재사용하기가 가능한가?
 

### 개선 제안
1. TabItem( Tab 항목을 클릭하는 ) 스타일을 좀 더 자유롭게 커스텀 할 수 있는 기능이 추가되었으면 좋겠다. 

```tsx
<Tabs scrollButtons={false} direction={"horizontal"} centered={true}>
  <Tabs.Panel
    style={{ backgroundColor: "red" }}
    label="별 하나의"
  >
</Tabs>
```
2. 라벨에 문자열이 없는 경우에 대한 처리
라벨에 문자열이 없을 경우 타입 에러나 컴파일 에러를 뱉으면 더 좋을 거 같습니다. 

3. 역시 라벨에 문자열 뿐만 아니라 이미지나 아이콘이 들어왔을 때도 대처를 할 수 있으면 더 좋을 거 같아요!!

**Reply by jeongwusi** [reply]
# 긍정적인 리뷰
## 여러 환경에서 잘 보이는가?
크롬, 파이어폭스, 사파리에서 확인을 해보았습니다. 여러 환경에서 시험을 해보았을 때, 잘 나오는 것을 확인할 수 있었습니다. 👍🏻👍🏻👍🏻

## 문서가 가독성 있게 쓰여있는가?
문서의 경우 사용자가 바로 적용할 수 있도록 잘 쓰여져 있었습니다. 다만 사용 예시에는 완전 기본적인 예시만 나와있는데, 각각의 Props에 맞는 예시도 함께 작성해주시면 더욱 빠르게 해당 라이브러리를 사용할 수 있을 거 같습니다.

## ‘유연’하게 재사용 하기 좋은가?
탭의 수가 많은 경우 스크롤을 통해 나머지 탭도 확인 가능하게 해주신 부분이 고민을 많이 하신 흔적이 느껴져서 매우 좋았습니다. 😆 

다만 아이콘과 문자열이 함께 들어가면 아이콘이 들어간 부분이 개행이 되네용...?! 이 부분은 어쩔 수 없는 거 같지만... 생각을 해보시면 좋을 거 같습니당 😊

# 개선 제안
매우 완벽하지만...! accentColor props의 default가 #3b82f6로 되어있더라구용! 물론 Hex 값을 이용해서 수정할 수 있는 부분은 매우 좋지만, 제 생각에는 색상을 미리 정의하여 사용자에게 선택지를 주는 방법도 좋을 거 같다는 생각을 하였습니당 😁

**Reply by geuntaek1013** [reply]
## 긍정적인 리뷰

1. `useTabs`, `useTabsScroll` - Navigation을 통해 어떤 label에 해당하는 Panel이 렌더링이 되는지 알 수 있다는 점이 인상 깊었습니다.
2. `ResizeObserver`을 통해 스크롤 기능을 control 하는 부분이 인상 깊었습니다.
3. 버티컬 탭을 지원해주는 부분이 좋았습니다.
4. 탭과 스크롤을 선택했을 때, 스타일과 애니메이션 적용 되는 부분이 좋았습니다.
5. npm 배포 문서와 README에 사용법과 설명이 잘 되어 있어 좋았습니다.
6. 크로스 브라우징이 잘 됩니다.

## 개선 제안 

1. panel label 개수가 많고, 길어져서 가로 화면 폭을 넘어 스크롤이 생겼을 때 컨텐츠가 살짝 짤려서 보이는 현상이 있습니다.
2. props가 조금 많아서, 이해하는데 조금 시간이 걸렸습니다. props drilling이 발생할 것 같아요.
3. 추후에 반응형만 추가하면 너무 좋을 것 같습니다.

**Reply by Creative-Lee** [reply]
## 긍정적인 코멘트
- 스크롤 버튼 디테일이 상당히 좋습니다.
- 탭의 양 끝단까지 스크롤하면 버튼 disable 처리 되는것도 지리네요
- Readme도 아주 좋네요
- npm 보고 예시코드 복붙해왔는데 이것도 너무 좋아요.


## 개선 제안
- 스크롤 버튼의 범위가 녹색만큼 컸으면 좋겠어요. 누르기 더 편해질듯용
![image](https://github.com/woowacourse/retrospective/assets/77152650/b832ec45-b2e5-496f-8522-0409aec5e17c)

- 요거 각 탭의 텍스트 드래그 안되게 해도 좋을 것 같아요.
![image](https://github.com/woowacourse/retrospective/assets/77152650/d8210c4d-d173-487f-be50-23974dda12ee)

- 라벨을 악의적으로 길게 해봤는데 적정 너비까지는 늘어나고? 줄 바꿈이 되는것도 좋아보이네요
![image](https://github.com/woowacourse/retrospective/assets/77152650/e138ddd0-c9eb-4cc8-bcd9-9e1ded4bade2)

- 라벨이 엄청 길어지면 스크롤 버튼도 가운데로 내려가요 스크롤 버튼은 탭 라인(상단)에 고정되었으면 좋겠어용 
![image](https://github.com/woowacourse/retrospective/assets/77152650/bdf83550-a844-43a5-91b4-e2555cb0eb23)

- 훠리즌탈 탭 일때 스크롤 버튼 얼라인이 살짝 불편해요 ㅋㅋㅋㅋ ㅡ자로 맞춰주세용~~~~~
![image](https://github.com/woowacourse/retrospective/assets/77152650/dc8045bf-1e87-4e73-90c1-2c9d56bdaffc)

- 스토리북 보고 생각해봤는데, **스크롤 버튼 항상 있는게** 더 좋을지도 모르겠어요.
넘길수 있는 방법이 탭 텍스트 드래그해서 영역잡아 넘기는 거밖에 없는데, 
2번 개선제안이 수정되면 이마저도 못하니깐 넘길 수 있는 방법이 아예 없을듯요. 
고로 스크롤 버튼이 항상 있는게 좋지 않을까 합니다
![image](https://github.com/woowacourse/retrospective/assets/77152650/7da7bfea-4eb0-40dc-9889-44c2e22857e2)

- 엑셀 시트에 '라벨에 아이콘을 넣는 경우를 고려했는가~?' 가 보여서 이모지 넣어봤는데 이모지 기준으로 줄바꿈이 되네용 구현부 궁금쓰
![image](https://github.com/woowacourse/retrospective/assets/77152650/f15ed8ff-69f3-40ea-8568-a43a8a914677)
 
- 개인적으로 border bottom + 라벨 color 변경 외에 라벨 백그라운드도 조정하고 싶어용 이거 가능한가요?

**Comment by WaiNaat** [comment]
# 참새 - Masonry Layout

https://github.com/woowacourse/layout-component/pull/58

**Reply by inyeong-kang** [reply]
### 여러 환경에서 잘 보이는가?
* 반응형: 모바일 환경에서도 이미지의 너비와 높이가 적절하게 보이고, 배치도 적절합니다!
* 크로스 브라우징: 크롬, 웨일에서 잘 작동합니다.
### 문서가 가독성 있게 쓰여있는가?
* 예시 코드가 자세하게 적혀있어서 사용법을 금방 익힐 수 있었어요.
### 유연하게 재사용 하기 좋은가?
* 커스텀하고 싶은 스타일도 자유롭게 추가로 넣을 수 있어서 확장성 있는 컴포넌트였습니다!
### 라벨 텍스트가 매우 긴 경우를 고려했는가?
* 텍스트 아이템에 대한 경우도 친절하게 제시되어 있어서 좋았습니다. (children에서 매우 긴 텍스트에 대해 overflow 설정을 해주도록 설명을 첨부해주어도 좋겠어요~)
---
정말 완벽한 사용성의 컴포넌트입니다!👍👍👏👏
예외적인 케이스(이미지 로딩이 실패한 경우)에 대해서 대처해봐도 좋을 것 같습니다.

**Reply by NaveOWO** [reply]
### 긍정적인 리뷰
- 커스터할 수 있는 속성들이 제한되어있어서 확실히 사용할 때 헷갈리지 않아서 좋아요! 어떤걸 해줘야하는지 명확해지는 것 같아요.
- item에 props를 주지않고 부모 컴폰너트에서 모든 일을 다 하게 해주는 것 도 사용성이 좋은 것 같아요!

### 개선제안
- 더보기 전 마지막 이미지들(?)을 보여줄 때 빈공간이 있는게 좀 어색했어요! 이미지를 모두 보여주되 더보기 위의 이미지들은 잘라주는 방식도 괜찮을 것 같아요!
- 리드미상으로는 이미지들이 끝났을 때 더보기 버튼이 사라지는지 확인하기 어려워요! 스토리북에 다양한 상황을 추가해줘도 좋을 것 같아요!!

참새 므찌네요!!!👍

**Reply by wzrabbit** [reply]
## 긍정적인 리뷰
전체적으로 컴포넌트가 깔끔했으며, 스토리북을 통해 컴포넌트를 설명하는 부분도 자세했습니다.

- 브라우저의 가로 길이를 줄임에 따라 가로 길이에 따라 보이는 열의 개수가 줄어드는 것으로 반응형에 대비하였으며 극단적으로 가로 길이를 줄였을 때에도 열 하나에 해당하는 이미지의 크기가 줄어드는 것으로 대비하였습니다. 반응형에 대한 대비가 잘 되어 있는 것 같습니다!
- `Chrome`, `Edge`, `Safari`, `Firefox`, `Whale` 다섯 개의 브라우저에서 테스트했으며 모두 일관된 결과를 보였습니다.
- 내부에 이미지로 된 배열만 넣으면 알아서 결과를 보여줘서 사용법도 간단했습니다.
- 이미지를 보여 주기 전 미리 랜덤 색상의 스켈레톤을 보여주어 사용자 입장에서 덜 어색해할 것 같습니다.

## 개선 제안
- `as` 프로퍼티가 무슨 역할을 하는지 스토리북 등의 문서에 설명이 되어 있으면 좋을 것 같아요!
- 더 보여줄 이미지가 없을 경우 "더 이상 표시할 이미지가 없습니다." 와 같이 별도의 안내 메시지를 "더 보기" 버튼 자리에 두는 것도 좋아보여요!
- 공원이 말씀하신대로 극단적으로 세로가 긴 이미지가 들어오는 상황 등에 대비해 어떻게 하면 더 이미지를 자연스럽게 보여줄 수 있을지 고민하시는 것도 좋을 것 같습니다

많이 배워갑니다 참새~

**Reply by ukkodeveloper** [reply]
### 1. 여러 환경에서 잘 보이는가

1. 크롬, 사파리에서 정말 잘 보입니다!!! 
2. fluid 까지 고려해서 반응형 만들어주신 게 인상깊었어요.

### 2. 문서가 가독성있게 쓰여있는가?

- 매우 가독성이 있습니다. 특히 스토리북에서 props 뿐만 아니라 속성마다 description으로 친절하게 설명되어 있는 부분이 좋았습니다.
- 뿐만 아니라 code sandbox를 통해 직접 실습할 수 있도록 해준 것이 좋았습니다.
- 다양한 예시를 통해 라이브러리를 어떻게 사용할 지 예상할 수 있도록 한 점이 좋았습니다.

### 3. 유연하게 재사용하기 좋은가?

- 적당히 유연하면서도 강제하는 부분이 정말 좋았습니다.
    - Masonry를 구성하는 props만 뚫어준 부분.
    - 하위 아이템에 지정해줘야하는 스타일을 props로 두지 않은 부분
    - as로 tag 종류를 고를 수 있게 한 부분

### **제안!**

Item 컴포넌트 하위에 Img 뿐만 아니라 텍스트도 고려하고 있어서 이 부분이 맞는 제안인지는 모르겠습니다. 그럼에도 img를 주로 사용한다는 가정하에 img 스타일에 *`style*={{ objectFit: 'cover' }}`를 별도로 넣어주어야 하는데, Item 컴포넌트에서 이를 강제하는 방법은 어떠신가요?? 

`Object-fit` 을 mdn에 찾아보았는데, masonry 에서 다른 `cover`외에 다른 속성값을 쓰는 상황이 없을 것 같다는 생각을 했습니다. 

https://developer.mozilla.org/ko/docs/Web/CSS/object-fit

### 추가

정말 참새 Masonry 미쳤어요!!!! 완전 프로젝트에 써보고 싶을 정도에요 🙂

**Reply by Gilpop8663** [reply]
## 여러 환경에서 잘 보이는가?

반응형: 모바일 환경, PC 환경 모두 동등하게 잘 보입니다. 또한 itemSize를 퍼센트로 준다면 PC와 모바일 모두 다른 사이즈로 동일한 갯수로 보이도록 설정할 수 있다는 점이 인상 깊었습니다.
크로스 브라우징: , 파이어폭스 브라우저, 네이버 웨일 브라우저, 브레이브 브라우저, 크롬 브라우저 등 모든 환경에서 동등하게 작동합니다. 다만 익스플로어는 지원하지 않는것으로 파악됩니다

<img width="1440" alt="image" src="https://github.com/woowacourse/retrospective/assets/80146176/ebf01481-bd0b-4f40-b43c-7e5d3e2dd2c7">


## 문서가 가독성 있게 쓰여있는가?

예시 코드가 스토리북에 잘 정리가 되어있고, Prop의 타입 또한 설명이 잘 되어 있어서 이해하기 편했습니다.

## 유연하게 재사용 하기 좋은가?

내부의 children에서 사용자가 상호작용 코드를 추가로 작성할 수 있어서 너무 좋았습니다. 재사용성 최고!

## 라벨 텍스트가 매우 긴 경우를 고려했는가?

텍스트의 경우에는 참새의 컴포넌트를 사용하는 개발자가 직접 처리를 하는 것이 적합하다고 생각이 들었고, 그렇게 답변도 들었습니다 


## 긍정적인 리뷰

PC, 모바일 모두 고려한 반응형이 잘 되는 점이 너무 좋았습니다. 그리고 사용자가 브라우저의 크기를 줄여도 느려지는 이슈 없이 정상 작동하는 점이 너무 좋았어요

## 개선 제안

itmeSize 25% 와 같이 주었을 때 4개가 보여질 것을 예상했는데 3개가 떠서 24%로 줘서 사용했었어요. column 값을 추가해서 3개를 입력하면 3개를 보여주는 prop도 있으면 좋을 것 같아요. 이렇게 되었을 때 itemSize와 개수 2개가 있으면 문제가 생길 것 같아서요.  적고 보니 이대로 itemSize만 지원하는 것이 사용자에게 헷갈리지 않을 것 같다는 생각이 들었어요

단 한가지 개선점으로 적어볼 만한 것으로 row 방향으로도 지원해주면 더욱 좋을 것 같아요!

**Comment by chsua** [comment]
# 부엉이 - Tab Layout
https://github.com/woowacourse/layout-component/pull/63

**Reply by 2yunseong** [reply]
className으로 지정을 해서 사용자에게 스타일을 위임을하고, 기능만 담당하는 레이아웃도 괜찮은 것 같습니다! 명시적으로 각 컴포넌트가하는 역할을 네이밍을 해주고, 컴포넌트 구조는 강제하고 스타일은 자유롭게 가져갈수 있는 것도 하나의 장점 같아요.

현재 스토리북에서도 할 수 있지만 defaultIdx를 설정을 두번째에 주고, 첫번째 컴포넌트에 1을 주어버리면 문제가 발생하네요! 

또한 tab의 children 갯수가 다를 때(아래 예시), 사용자가 레이아웃 배치를 커스텀할 수 있도록 TabList의 스타일을 직접 커스텀 할 수 있게 하면 좋을 것 같습니다 :)
<img width="236" alt="image" src="https://github.com/woowacourse/retrospective/assets/56749516/41ea132e-00cd-4609-ab2f-d4b07e04c242">

레포지토리에 있는 README 내용도 npm 에 있는대로 수정하면 좋을 것 같고, Tab, TabList등에도 이 컴포넌트가 무슨 일을 하는지 설명이 있으면 좋을 것 같아요 !! :D

**Reply by chsua** [reply]
## 긍정적인 리뷰
- 심플해서 좋습니다.
- 저도 탭 레이아웃을 했는데 탭 레이아웃과 탭을 분리하여 내보내는 건 생각하지 못했습니다. 저는 해당 부분을 객체로 입력하도록 했는데 가독성 측면에서 이렇게 하는게 더 나을 수 있다고 생각이 들었습니다. 

## 개선 제안
- 의도한 기능(class-name을 이용하여 style custom, 페이지 렌더링 시 자동 포커즈 등)을 정리해서 보여주면 좋을 것 같아요.! 그와 함께 props에 대한 자세한 설명이나 의도가 드러나면 좋을 것 같아요.
- 라이브러리에서 불러오는 태그(tabs, tab, tabList...)가 여러 개라 전체적인 사용법(사용예시)가 read me에 있으면 더 이해하기 쉬울 것 같아요.

**Reply by regularPark** [reply]
## 리뷰🦉 

- 구조적으로 깔끔하게 잘 만드신 것 같아요.
- 리드미도 쉽게 잘 읽혔어요! 
- Context를 잘 활용하셨네요😁

### 개선 제안
- 사용자에게 스타일의 자유도를 많이 주셨군요! 하지만 부엉이의 스타일을 드러내보는 것은 어떨까요~?
- `TabPanel` 내의 `activeIdx`가 `number` | `string`이어서 `index`와 `name`이 동시에 동일한 케이스가 있어서 `TabPanel`이 2개 이상 보이는 케이스를 발견했어요!🥲

**Reply by n0eyes** [reply]
- 사용하는 속성들이 직관적이었어요!
- 사용하는 개발자의 DX가 좋았어요!
----

- 탭 전환을 위해 버튼과 패널을 연결하는 `name` 속성을 사용하고 있으니까 `defaultIdx` 속성도 `name`으로 통일하면 더 좋을 것 같아요!
- 접근성 속성이 정상적으로 동작하지 않는 것이 있었어요! (aria-control - id)

**Reply by tkdrb12** [reply]
## 긍정적인 리뷰
 - ref를 활용해 탭을 컨트롤 하는 부분이 인상적이었습니다! 제 팀 프로젝트 코드에도 이 방식을 활용해 리팩토링 해보고 싶어요  
 - className을 통해 임의로 스타일을 커스텀 할 수 있어 확장성이 높은 것 같아요.

## 개선 제안
 - className을 이용하여 스타일을 임의로 부여하는 부분의 확장성은 좋지만 설명이 부족한 것 같아요. README에 사용 예시나 설명을 좀 더 구체적으로 적어주시면 좋을 것 같아요
 - 스토리북에서는 autoFocus를 사용하기 힘들어 헷갈릴 수 있다고 생각해요 이 것도 좀 더 설명을 구체적으로 적어주시면 좋을 것 같아요

**Comment by kangyeongmin** [comment]
# 패트릭 - Drawer

https://github.com/GC-Park/layout-component/tree/step2

**Reply by kangyeongmin** [reply]
## 긍정적인 리뷰

1. 일단 리드미 작성이 너무 잘되있어서 한 눈에 이해가 갔어요. 
2. 속성들을 테이블 형태로 작성하신 점이 좋아서 저도 다음에 그렇게 하려구요.
3. 필요한 속성들만 간단하고 알차게 구현하신 점이 좋았어요.

## 개선제안

1. size가 full 일때, 닫을 수 없어요. drawer의 개념이다보니 제일 큰 사이즈도 딤드된 영역을 조금 남기도록 하면 어떨까요? 
2. 사이즈를 추상적인 단위보다는, 사용자의 환경에 맞게 세밀하게 조절가능 하도록 px이나 rem 단위도 취급해주면 훨씬 더 좋은 라이브러리가 될 것 같아요.
3. 사용처에서 useState와 여는 함수, 닫는 함수를 선언하고 있는데, useDrawer이라는 커스텀 훅을 만들어서 라이브러리에서 뽑아 쓸 수 있게 하면 어떨까요?
4. closeSheet이라는 네이밍을 바꿔보면 어떨까요? `sheet`과 `drawer`라는 이름을 통일시켜주면 좋을 것 같아요!
5. 배경색이 불투명한데 이것도 속성으로 사용자가 조절할 수 있게 하면 어떨까요?

**Reply by yeopto** [reply]
# 긍정적인 리뷰
1. 리드미 설명이 너무 좋았어용!
2. 열릴 때 애니메이션 있어서 참 좋았네요~

# 개선제안
1. Drawer는 Modal처럼 전역에서 사용하는 컴포넌트라 생각해요. 지금 상태로는 부모 컴포넌트에 종속적일 수 있다고 생각합니다. 사용자가 예상하지 못했던 결과가 나올 수 있을거 같아요.
2. 반응형이 안되어있는데요. 반응형을 따로 한다기 보단 portal을 이용하고 Contents영역의 width를 children의 width로 가져가면 1번과 2번 둘다 해결할 수 있지 않나 싶네요!
3. 1번 2번에 이어서 size prop이 꼭 필요할까 싶습니다! 라잇의 말대로 배경색을 사용자가 조절할 수 있게 해주는 옵션이 있으면 좋을 것 같네용!

**Reply by semnil5202** [reply]
## 긍정적인 리뷰
* 리드미 작성이 야무지네요.
* 사이즈를 사용자가 직접 px 단위로 지정하지 않고 'sm', 'xs' 등과 같은 단위를 입력하게 함으로써 조금 더 편리한 것 같아요. 옵션 값도 많기 때문에 대부분의 니즈를 충족할 수 있을 것 같아요.

## 개선 제안
* closeSheet 속성에 닫은 함수를 넣어줘야 dimmed 된 영역을 눌러 닫을 수 있다는 플로우가 잘 와 닿지 않아요. 속성 네이밍 변경 또는 closeSheet를 넘겨주지 않아도 닫을 수 있도록 플로우를 변경해보면 어떨까요?
* 배경색을 사용자가 조정해볼 수 있게 변경하면 더 좋을 것 같아요.

**Reply by bassyu** [reply]
## 긍정적인 리뷰

- 아래 뿐만 아니라 위, 양옆으로 설정할 수 있어서 좋았어요.
- 여닫는 동작을 밖에서 제어할 수 있도록 해서 좋았어요.

## 개선제안

- size를 자유롭게 해도 좋을 것 같아요! UI 라이브러리들은 거의 필요한 모든 UI가 들어있고, 통일성을 주기 위해 size의 역할이 크다고 생각했어요.
반면 저희 라이브러리는 기존의 UI에 추가로 들어가는 입장이라, 오히려 자유로우면 좋겠다고 생각했습니다~
- left, right position을 auto로 해서 그런지 왼쪽에 여백이 생기는데 여백이 없어야 할 것 같습니다!

**Comment by nangkyeonglim** [comment]
# 팀 해온의 신나는 TabLayout 리뷰 시간 🥳

https://hae-on.github.io/component-docs/docs/category/layout/

## 긍정적 리뷰

### 파인

여러 스타일을 각각 설정할 수 있어 좋다. 참고할만한 좋은 설정이다

### 황펭

children을 배열을 이용해서 여러 tab들을 만들 수 있게 명확하게 알려줘서 좋다. active, inactive 스타일을 커스텀할 수 있어 좋다.

### 타미

사용자들이 쓸 때 쉽게 커스터마이징 할 수 있을 것 같아서 좋다.

### 루루

탭 컴포넌트가 탭으로 움직일 수 있다는 것이 좋다. 적용해보고 싶다.

### 아커

커스텀 자유도가 높아서 좋은 것 같다. 디자인도 예쁘다.

## 개선 사항 제안

### 파인

접근성을 고려해서 aria 속성들을 고려해보는 것이 어떤지 제안합니다.

### 황펭

탭 이동할 때 애니메이션이 조금 있으면 좋을 것 같습니다. 애니메이션 선택까지 있다면 더 좋을 것 같습니다.

### 타미

tabListStyles안에 activeStyle과 inactiveStyle이 있는데 문서에 계층을 분리해주면 좀더 좋을 것 같습니다.

### 루루

탭과 container(탭과 내용)이 좀더 명확하게 연결되어있는 것이 보이면 좋겠습니다.

### 아커

현재 모바일 환경에서 탭이 잘리게 보이는데 반응형을 좀 더 고려하면 좋을 것 같습니다.

**Reply by xodms0309** [reply]
좋네요..

**Comment by yogjin** [comment]
# 가브리엘🩳과 함께하는 SplitPane 리뷰!
https://github.com/woowacourse/layout-component/pull/62

## 긍정적 리뷰
1. 퍼센트로 받아서 직관적이라 좋았습니다.
2. 자연스러운 드레그 기능이 인상적입니다.
3. 프롭스 이름이 직관적이라 알아보기 쉽습니다
4. 사용성이 매우 좋은것 같아요! 레이아웃 하나 놓고 자식만 넣으면 자동으로 나뉘는게 매우 좋습니다!

## 개선 사항 제안
1. 퍼센트를 string대신 숫자로 받아서 내부적으로 퍼센트 처리하면 좋을 것 같아요
2. divider를 조금 더 넓게 해주면 조작하기 더 편할 것 같습니다.
3. 가로 기능도 만들어주시면 좋을 것 같습니다.
![image](https://github.com/woowacourse/retrospective/assets/33623078/e7e3a3ec-8175-4f0c-bf9d-38d79d102c44)
4. props에 대한 설명이 더욱 자세했으면 좋았을 것 같아요!(에디터에 마우스를 오버했을 때)
5. resizer 색깔이 검은색이면 좋겠어요(지금 하얀색)

**Comment by HyeryongChoi** [comment]
# 제레미's Parallax-scrolling 리뷰👏
https://github.com/woowacourse/layout-component/pull/51

**Reply by ksone02** [reply]
## 긍정적인 부분
- Perspective, translateZ등의 css property를 활용해서 실제 물리학적인 관점의 차이에 의해 발생하는 속도차이를 구현한 것 같아서 좋았다.
- README가 상세하게 작성되어있었다. 특히 구조 및 스타일링, 결론과 같은 파트를 작성해주어 컴포넌트의 이해도가 올라간 것 같다.

## 개선 제안
- ParallaxScrolling에 대한 story를 비롯한 예시를 보고 싶었는데, 없어서 아쉽다.
- ParallaxItem의 html 태그가 img로 국한되어있어서 아쉽다. 이미지를 넣고싶을 수도 있고, 텍스트를 넣고 싶을 수도 있고, 또 다른 Html엘리먼트를 넣고 싶을 수도 있는데 고려가 안된 것 같다.
- perspective와 translateZ로 조절하고 있는데, speed로 조절하는 방식이 사용자 입장에선 더 편하지 않았을까? 실제 물리법칙에 의해 움직이기 때문에, 오히려 사용자의 의도와 다른 결과를 가져올 수 있으니 Item의 speed를 커스텀할 수 있게 제공하는게 낫지 않았을까…?

**Reply by HyeryongChoi** [reply]
## 긍정적인 리뷰
예시코드를 npm readme에 써줘서 테스트해보기 편했습니다👍

## 개선제안
```tsx
<ParallaxScrolling perspective={0}>
        <ParallaxItem
          translateZ={-10}
          scale={23}
          imgUrl='https://d3ex8zhlkii399.cloudfront.net/galaxy.jpg'
          objectFit='cover'
        />
        <ParallaxItem
          translateZ={-10}
          scale={3}
          imgUrl='https://d3ex8zhlkii399.cloudfront.net/neptune.png'
          right='10000px'
          objectFit='contain'
        />
        <ParallaxItem
          translateZ={-50}
          scale={10}
          imgUrl='https://d3ex8zhlkii399.cloudfront.net/mars.png'
          objectFit='contain'
          left='8000px'
        />
</ParallaxScrolling>
```
배경과 같은 translateZ(-10)를 주면 같은 속도로 스크롤이 되고 배경보다 더 작은 값(-50)을 주면 배경이 이미지보다 느리게 스크롤이 되네여
근데 이미지의 사이즈도 이 값에 따라 작아지고 커지고 해서 헷갈립니다! 속도를 조절할 수 있는 옵션이 있으면 더 좋을 것 같아요👍

**Reply by suyoungj** [reply]
### 긍정적인 면
- 스토리북 문서화가 잘 되어 있었고, 예시 페이지를 통해 잘 동작하는 것을 쉽게 확인할 수 있었습니다.
- 필요한 prop과 각각에 대한 설명이 잘 되어 있어서 문서만 읽고 컴포넌트를 바로 사용하는 것이 가능했습니다.

### 개선 제안
- props가 사용하는 입장에서 진입장벽처럼 느낄 수 있을 것 같습니다. speed와 같은 prop으로 간편하게 속도를 조절할 수 있다면 더 직관적이고 사용하기 좋을 것 같아요.
- 스토리북 이외에 전체 페이지로 된 예시를 볼 수 있다면 더 좋을 것 같습니다.
- `perspective, translateZ`값을 조금만 변경해도 요소의 크기가 극심하게 변해서 사용자 입장에서 놀랄 수 있을 것 같습니다.

**Reply by jiwonh423** [reply]
긍정적인 부분
npm README가 처음 사용하는 사람에게 눈높이를 맞춰 꼼꼼히 설명해줘서 좋았습니다. 
개선 제안
ParallaxScrolling이 애플 랜딩페이지처럼 더 다이나믹한 기능이 있는 줄 알았는데 단순한 조절만 가능해서 아쉬웠습니다.
ParallaxItem들에 대해서도 더 동적으로 조절가능하면 좋았을 것 같습니다.(어렵겠지만)

**Reply by guridaek** [reply]
### 긍정적인 부분
- 이게 어떤 컴포넌트인지 스토리북에서  잘 보여주는 것 같아요.
### 개선 제안
- translateZ, scale 등 Item의 props들의 역할을 이해하기가 조금 힘든 것 같아요.

**Comment by solo5star** [comment]
## 솔로스타 - Tab Component
https://github.com/woowacourse/layout-component/pull/61

차린게 없습니다

**Reply by solo5star** [reply]
## 긍정적인 부분
* 문서가 단순하고 보기 좋았다
* 사용법이 직관적이고 좋았다
* 컴포넌트가 직관적이고 사용하기가 정말 쉬웠어요.

## 개선 제안
* css도 함께 import해줘야 한다는 점이 아쉬웠어요.
* 각 탭 버튼의 스타일도 지정할 수 있으면 좋을 것 같아요.
* 너비 조절도 가능하면 좋을 것 같아요
* 탭 버튼의 텍스트가 활성화/비활성화될 때 위 아래로 와리가리? 한다
